File: cpp/13b_734A_AntonAndDanik.cpp
#include<bits/stdc++.h>

int main(){
    int n;
    std::string winners;
    std::cin>>n>>winners;
    std::unordered_map<char,int> hmap;
    for (auto i : winners){
        hmap[i]++;
    }
    if(hmap['A'] > hmap['D']){
        std::cout<<"Anton";
    }
    else if (hmap['D'] > hmap['A']){
        std::cout<<"Danik";
    }
    else{
        std::cout<<"Friendship";
    }
    return 0;
}

File: python/13b_734A_AntonAndDanik.py
def main():
    n = int(input())
    winners = input().rstrip()
    hmap = {"A" : 0,
            "D" : 0}
    for i in winners:
        hmap[i] += 1
    if hmap["A"] > hmap["D"] :
        print("Anton")
    elif hmap["A"] < hmap["D"] :
        print("Danik")
    else:
        print("Friendship")

if __name__ == "__main__":
    main()